### PR TITLE
[API] Update CapabilitiesResponse.displayFields to be an array

### PR DESCRIPTION
**Pull request by bfcrampton**
This is needed in order to maintain priority ordering of the fields. An `object` (or Map) does not have  guaranteed ordering.

### CODE REVIEW

File: api/jobs.yaml [hunk 1]
```diff
@@ -165,6 +167,11 @@ definitions:
         description: Fields on ExtendedQueryFields which are queryable
         items:
           type: string
+  DisplayField:
+    description: Map of field name to display name
```
Comment (calbach) [inline api/jobs.yaml]: ultranit: I'd say description of a display field; could extend this with more later potentially

File: api/jobs.yaml [hunk 2]
```diff
@@ -201,85 +208,12 @@ definitions:
       labels:
         type: object
         description: Custom job labels with string values
```
Comment (calbach) [inline api/jobs.yaml]: How'd all this other stuff move?

File: api/jobs.yaml (hunk 1 above)
Comment (bfcrampton) [reply api/jobs.yaml]: Done.

File: api/jobs.yaml (hunk 2 above)
Comment (bfcrampton) [reply api/jobs.yaml]: Oops I copied some stuff from my PR with job models updates. Removed.

**Pull request merged**